app/views/events/index.html.erb

```diff
@@ -3,22 +3,39 @@
 <h1>Events</h1>
 <% if current_user && current_user.admin %>
   <%= link_to 'New Event', new_event_path, class: "btn btn-secondary" %>
+  <% visible_events = @events %>
+<% else %>
+  <% visible_events = @events.select {|event| event.group.users.exists?(current_user.id) && event.published && event.eventType != "message"} %>
 <% end %>
-<% @events.each do |event| %>
+<% visible_events.each do |event| %>
```

Comment:
I'm working on displaying this file nicer, so I guess we can merge this and then I'll merge it into mine. We can merge this after the changes, and then I want to put the event cards into a partial.